fmskill/comparison.py

```diff
@@ -1080,124 +933,165 @@ def scatter(
         )
         return ax
 
-    def taylor(
-        self,
-        model: Union[str, int, List[str], List[int]] = None,
-        observation: Union[str, int, List[str], List[int]] = None,
-        variable: Union[str, int, List[str], List[int]] = None,
-        start: Union[str, datetime] = None,
-        end: Union[str, datetime] = None,
-        area: List[float] = None,
-        df: pd.DataFrame = None,
-        normalize_std: bool = False,
-        aggregate_observations: bool = True,
-        figsize: List[float] = (7, 7),
-        marker: str = "o",
-        marker_size: float = 6.0,
-        title: str = "Taylor diagram",
-    ):
-        """Taylor diagram showing model std and correlation to observation
-        in a single-quadrant polar plot, with r=std and theta=arccos(cc).
 
-        Parameters
-        ----------
-        model : (int, str), optional
-            name or id of model to be compared, by default all
-        observation : (int, str, List[str], List[int])), optional
-            name or ids of observations to be compared, by default all
-        variable : (str, int), optional
-            name or id of variable to be compared, by default first
-        start : (str, datetime), optional
-            start time of comparison, by default None
-        end : (str, datetime), optional
-            end time of comparison, by default None
-        area : list(float), optional
-            bbox coordinates [x0, y0, x1, y1],
-            or polygon coordinates[x0, y0, x1, y1, ..., xn, yn],
-            by default None
-        df : pd.dataframe, optional
-            show user-provided data instead of the comparers own data, by default None
-        normalize_std : bool, optional
-            plot model std normalized with observation std, default False
-        aggregate_observations : bool, optional
-            should multiple observations be aggregated before plotting
-            (or shown individually), default True
-        figsize : tuple, optional
-            width and height of the figure (should be square), by default (7, 7)
-        marker : str, optional
-            marker type e.g. "x", "*", by default "o"
-        marker_size : float, optional
-            size of the marker, by default 6
-        title : str, optional
-            title of the plot, by default "Taylor diagram"
+class SingleObsComparer(BaseComparer):
+    def __init__(self, observation, modeldata=None):
```

Comment:
Why is modeldata an optional argument? What am I supposed to compare if I dont have any modeldata?